godot-macros/src/docs.rs

```diff
@@ -39,24 +40,41 @@ pub fn make_inherent_impl_docs(
     constants: &[ConstDefinition],
     signals: &[SignalDefinition],
 ) -> TokenStream {
+    /// Generates TokenStream containing field definitions for documented functions, constants and signals.
+    /// Returns None if none of above has been documented
     fn pieces(
         functions: &[FuncDefinition],
         signals: &[SignalDefinition],
         constants: &[ConstDefinition],
     ) -> Option<TokenStream> {
         let methods = functions
             .iter()
-            .map(make_method_docs)
-            .collect::<Option<String>>()?;
+            .filter_map(make_method_docs)
+            .collect::<String>();
         let signals = signals
             .iter()
-            .map(make_signal_docs)
-            .collect::<Option<String>>()?;
+            .filter_map(make_signal_docs)
+            .collect::<String>();
         let constants = constants
             .iter()
             .map(|ConstDefinition { raw_constant: x }| x)
-            .map(make_constant_docs)
-            .collect::<Option<String>>()?;
+            .filter_map(make_constant_docs)
+            .collect::<String>();
+
+        if methods.is_empty() && signals.is_empty() && constants.is_empty() {
+            return None;
+        }
+        let to_tokenstream = |s: String| -> TokenStream {
+            s.is_empty()
+                .not()
+                .then(|| quote!(Some(#s)))
+                .unwrap_or(quote! {None})
+        };
+        let (methods, signals, constants) = (
+            to_tokenstream(methods),
+            to_tokenstream(signals),
+            to_tokenstream(constants),
+        );
```

Comment:
Why are you doing this `s.is_empty().not().then().unwrap_or_else()` thing?
```suggestion
        let to_tokenstream = |s| match s {
           "" => quote! { Some(#s) },
           _ => quote! { None },
        };
        let (methods, signals, constants) = (
            to_tokenstream(methods),
            to_tokenstream(signals),
            to_tokenstream(constants),
        );
```

Reply:
fixed